src/store/modules/nft.js

```diff
@@ -939,17 +942,37 @@ const actions = {
     { commit },
     { collectionId, coupon }
   ) {
-    const {
-      data: { fiatPrice, LIKEPrice },
-    } = await this.$api.get(
-      api.getNFTBookPaymentPrice({ collectionId, coupon })
-    );
-    const info = { fiatPrice, LIKEPrice };
-    commit(TYPES.NFT_SET_NFT_CLASS_PAYMENT_PRICE_INFO, {
-      collectionId,
-      info,
-    });
-    return info;
+    const fetchPaymentInfo = params =>
+      this.$api
+        .get(api.getNFTBookPaymentPrice(params))
+        .then(response => response.data);
+
+    const requests = [fetchPaymentInfo({ collectionId })];
+    if (coupon) {
+      requests.push(fetchPaymentInfo({ collectionId, coupon }));
+    }
```

Comment:
Refactor after https://github.com/likecoin/likecoin-api-public/pull/654 is merged